packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -9,7 +9,14 @@ import { Disposable, Event } from '../../common';
 
 export const PreferenceProvider = Symbol('PreferenceProvider');
 
+export enum Scope {
+    User,
+    Workspace
+}
+
 export interface PreferenceProvider extends Disposable {
+    getScope?(): Scope;
```

Comment:
It should be handled it by named injections in `PreferenceService`, i.e.
```ts
@inject(PreferenceProvider) @named(PreferenceScope.User)
protected readonly userPreferenceProvider: PreferenceProvider
```

Reply:
done